modules/hal_nordic/nrfx/CMakeLists.txt

```diff
@@ -134,3 +137,15 @@ if(DEFINED uicr_path)
     zephyr_library_compile_definitions(CONFIG_GPIO_AS_PINRESET)
   endif()
 endif()
+
+if(CONFIG_SOC_NRF54L15)
+	dt_prop(clock_frequency PATH "/cpus/cpu@0" PROPERTY "clock-frequency")
+	math(EXPR clock_frequency_mhz "${clock_frequency} / 1000000")
+	zephyr_compile_definitions("CONFIG_CPU_FREQ_MHZ=${clock_frequency_mhz}")
```

Comment:
`CONFIG_` is used by Kconfig and exported to code through `autoconf.h`.

Why are we taking a devicetree property into CMake, do math ops on its value, and then setting it as a compile definition which appears as being a Kconfig ?

Reply:
It looks like a definition required by HAL headers? but with NRF_CONFIG_ prefix?

Reply:
It's an MDK config define, used to be `CONFIG_CPU_FREQ_MHZ`, recently renamed to `NRF_CONFIG_CPU_FREQ_MHZ`:
https://github.com/zephyrproject-rtos/hal_nordic/blob/master/nrfx/mdk/system_nrf54l.c#L60
So this line should stay, but updated with new name.

Reply:
Updated